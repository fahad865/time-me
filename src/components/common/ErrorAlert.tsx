import { Alert } from 'antd';
import * as React from 'react';
import { AppError } from '../../types';

export interface Props {
  error: AppError;
  errorReset: () => void;
}

class ErrorAlert extends React.Component<Props, {}> {
  constructor(props: Props) {
    super(props);
  }

  render() {
    const { error } = this.props;
    return (
      <div style={{ margin: '5px' }}>
        {
          error.errorMessage ?
            <Alert
              type="error"
              banner={false}
              closable={true}
              showIcon={true}
              afterClose={this.props.errorReset}
              message={error.errorMessage}
            />
            : null
        }
      </div>
    );
  }
}

export default ErrorAlert;