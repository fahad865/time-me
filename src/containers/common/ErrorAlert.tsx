import ErrorAlert from '../../components/common/ErrorAlert';
import { StoreState, AppError } from '../../types';
import { connect, Dispatch } from 'react-redux';
import * as actions from '../../actions/errorActions';

export type StateFromProps = {
  error: AppError;
};

export type DispatchFromProps = {
  errorReset: () => void;
};

export function mapStateToProps({ error }: StoreState) {
  return {
    error
  };
}

export const mapDispatchToProps = (dispatch: Dispatch<AppError>): DispatchFromProps => ({
  errorReset: () => dispatch(actions.errorReset()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ErrorAlert);