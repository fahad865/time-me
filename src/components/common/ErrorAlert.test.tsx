import * as React from 'react';
import { shallow, mount } from 'enzyme';
import ErrorAlert from './ErrorAlert';

function setupShallow(errorMessage: string) {
  const props = {
    error: {
      errorMessage: errorMessage
    },
    errorReset: jest.fn()
  };

  const enzymeWrapper = shallow(<ErrorAlert {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

function setup(errorMessage: string) {
  const props = {
    error: {
      errorMessage: errorMessage
    },
    errorReset: jest.fn()
  };

  const enzymeWrapper = mount(<ErrorAlert {...props} />);

  return {
    props,
    enzymeWrapper
  };
}

describe('Error Alert', () => {
  it('should render correctly with message', () => {
    const { enzymeWrapper } = setupShallow('Error');
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should render correctly without message', () => {
    const { enzymeWrapper } = setupShallow('');
    expect(enzymeWrapper).toMatchSnapshot();
  });

  it('should call errorReset when alert is closed', () => {
    const { enzymeWrapper, props } = setup('Error');
    const alert = enzymeWrapper.find('Alert').instance() as any;
    if (alert.props.afterClose) {
      alert.props.afterClose();
      expect(props.errorReset.mock.calls.length).toBe(1);
    }
  });
});