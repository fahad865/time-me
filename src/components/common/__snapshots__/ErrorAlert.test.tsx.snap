// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Alert should render correctly with message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorAlert
    error={
        Object {
            "errorMessage": "Error",
          }
    }
    errorReset={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Alert
        afterClose={[Function]}
        banner={false}
        closable={true}
        message="Error"
        showIcon={true}
        type="error"
/>,
      "style": Object {
        "margin": "5px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "afterClose": [Function],
        "banner": false,
        "closable": true,
        "message": "Error",
        "showIcon": true,
        "type": "error",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Alert
          afterClose={[Function]}
          banner={false}
          closable={true}
          message="Error"
          showIcon={true}
          type="error"
/>,
        "style": Object {
          "margin": "5px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "afterClose": [Function],
          "banner": false,
          "closable": true,
          "message": "Error",
          "showIcon": true,
          "type": "error",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Error Alert should render correctly without message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ErrorAlert
    error={
        Object {
            "errorMessage": "",
          }
    }
    errorReset={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": null,
      "style": Object {
        "margin": "5px",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "style": Object {
          "margin": "5px",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
